test/test_generate_bed.py

```diff
@@ -0,0 +1,200 @@
+"""
+Tests for the `generate_bed.py` script in the PanelPal module using
+`pytest` and `unittest.mock`.
+
+Tests include:
+
+- Handling of missing or incomplete arguments
+- Handling of invalid genome build
+- Successful execution with valid arguments
+- Exception handling for various error scenarios during execution
+- Mocking logger to simulate errors and validate error handling
+
+Fixtures:
+- `mocked_logger_fixture`: Mocks the logger to capture output during tests.
+"""
+
+import subprocess
+import sys
+from unittest import mock
+import pytest
+from accessories import variant_validator_api_functions, panel_app_api_functions
+from ..PanelPal.generate_bed import main
+
+def test_missing_required_arguments():
+    """
+    Test script behavior when all arguments are missing.
+    """
+    result = subprocess.run(
+        [sys.executable, "PanelPal/generate_bed.py"],
+        capture_output=True,
+        text=True,
+        check=False
+    )
+    assert result.returncode != 0
+    assert "the following arguments are required" in result.stderr
+
+
+def test_missing_single_argument():
+    """
+    Test script behavior when a single argument is missing.
+    """
+    result = subprocess.run(
+        [sys.executable, "PanelPal/generate_bed.py", "-p", "R207", "-v", "4"],
+        capture_output=True,
+        text=True,
+        check=False
+    )
+    assert result.returncode != 0
+    assert "the following arguments are required: -g/--genome_build" in result.stderr
+
+def test_invalid_genome_build():
+    """
+    Test script behavior with invalid genome build.
+    """
+    result = subprocess.run(
+        [
+            sys.executable,
+            "PanelPal/generate_bed.py",
+            "-p", "R207",
+            "-v", "4",
+            "-g", "INVALID_GENOME"
+        ],
+        capture_output=True,
+        text=True,
+        check=False
+    )
+    assert result.returncode != 0
+    assert "invalid choice: 'INVALID_GENOME' (choose from 'GRCh37', 'GRCh38')" in result.stderr
+
+def test_valid_arguments():
+    """
+    Test script behavior with valid arguments.
+    """
+    # Run the script as a subprocess
+    result = subprocess.run(
+        [
+            sys.executable,
+            "PanelPal/generate_bed.py",
+            "-p",
+            "R207",
+            "-v",
+            "4",
+            "-g",
+            "GRCh38",
+        ],
+        capture_output=True,
+        text=True,
+        check=False
+    )
+
+    # Assert successful execution
+    assert result.returncode == 0
+
+    # Check for the success message in either stdout or stderr
+    output = result.stdout + result.stderr
+    assert "Process completed successfully" in output
```

Comment:
I wonder if it's possible, but also i don't know if neccessary, to include a test where you can provide what the output file would be for a given set of inputs. Perhaps a panel with just one gene with few exons. I'm not sure where that file would be stored. But I'm sure it would be possible.

Reply:
I have amended this function to test that the files have been generated and that they are not empty for now. It was trickier than expected as I wanted to move the output files to a tmp folder so they could be deleted after testing. 

![image](https://github.com/user-attachments/assets/6599331c-51e7-4fbb-bc5c-4183a63b61ac)


I think the testing of the file contents can be improved after the sprint 4 changes to the bed file (i.e., after the concatenation of the final columns and when there will be a hash out header).

I would also like generate_bed.py to have an option to specify an output file destination folder so they aren't just saved to the current working directory. But this can be a future enhancement.